applications/segyio-catb.c

```diff
@@ -210,20 +149,41 @@ int main( int argc, char** argv ){
 
         if( err ) return errmsg(opterr, "Unable to read binary header");
 
-        for( int c = 0; c < 30; ++c ){
-            int field = get_binary_value( binheader, bfield_value[ c ] );
-            if( opts.nonzero && !field) continue;
-
-            if( opts.description ) {
-                int byte_offset = (bfield_value[ c ] - SEGY_TEXT_HEADER_SIZE);
-                printf( "%s\t%d\t%d\t%s\n",
-                    su[ c ],
-                    field,
-                    byte_offset,
-                    su_desc[ c ] );
+        int nr_fields = sizeof(field_data)/sizeof(binary_field_type);
+        for( int c = 0; c < nr_fields; ++c ){
+            segy_field_data fd = segy_get_field( binheader, field_data[c].offset );
+            if ( fd.error ) return errmsg( fd.error, "Unable to read field" );
+
+            if( opts.nonzero && fd.value.u32 == 0) continue;
```

Comment:
Oh, so this is the problem... Honestly missed this line when reading.

Maybe it has something to do with environments and compilers, after you removed u64=0 assignment?
Anyway it would look better inside datatype "if"s, so good that CI noticed that.